addons/test_website/tests/test_views_during_module_operation.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import TransactionCase, tagged
+
+
+@tagged('-at_install', 'post_install')
+class TestViewsDuringModuleOperation(TransactionCase):
+    def test_01_cow_views_unlink_on_module_update(self):
+        """ Ensure COW views are correctly removed during module update.
+        Not removing the view could lead to traceback:
+        - Having a view A
+        - Having a view B that inherits from a view C
+        - View B t-call view A
+        - COW view B
+        - Delete view A and B from module datas and update it
+        - Rendering view C will crash since it will render child view B that
+          t-call unexisting view A
+        """
+
+        View = self.env['ir.ui.view']
+        Imd = self.env['ir.model.data']
+
+        update_module_base_view = self.env.ref('test_website.update_module_base_view')
+        update_module_view_to_be_t_called = View.create({
```

Comment:
why not move all the creation logic to test_website_data.xml? 
with more generic name that "update_module_..."

Reply:
I cant add these views in xml file or it wont be considered as deleted on module update as explained in the comment